exchanges/binance/binance.go

```diff
@@ -17,63 +17,2476 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/common/crypto"
 	"github.com/thrasher-corp/gocryptotrader/currency"
 	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
 	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
 	"github.com/thrasher-corp/gocryptotrader/log"
 )
 
+// Binance is the overarching type across the Binance package
+type Binance struct {
+	exchange.Base
+	USDTWS              stream.WebsocketConnection
+	CoinMarginFuturesWS stream.WebsocketConnection
+	// Valid string list that is required by the exchange
+	validLimits []int
+}
+
 const (
-	apiURL = "https://api.binance.com"
+	apiURL         = "https://api.binance.com"
+	spotAPIURL     = "https://sapi.binance.com"
+	cfuturesAPIURL = "https://dapi.binance.com"
+	ufuturesAPIURL = "https://fapi.binance.com"
+
+	// Public endpoints
+
+	// Futures
```

Comment:
Just a "small" suggestion... split up the functionality to the target areas. I.e. create extra files
spot.go
margin.go
cfutures.go
ufutures.go
where you can keep the related stuff together. In golang the number of files in the package doesn't make any difference functionality wise, always feel free to add new files.
It'll be much easier to read the code and find any relevant functionality later on.
Maybe, prefix with `binance_` if you really want to preserve the package in the file name. (I personally prefer without as the package and the folder already provide this context)

Reply:
Have done the requested changes except for margin.go since margin is a part of spot